controllers/configmap_controller.go

```diff
@@ -414,6 +436,12 @@ func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			builder.WithPredicates(outdatedWindowsNodePredicate(true))).
 		Watches(&core.Node{}, handler.EnqueueRequestsFromMapFunc(r.mapToServicesConfigMap),
 			builder.WithPredicates(windowsNodeVersionChangePredicate())).
+		Watches(&core.Node{}, handler.EnqueueRequestsFromMapFunc(r.mapToServicesConfigMap),
+			builder.WithPredicates(nodeLabelChangedPredicate())).
+		Watches(&core.Node{}, handler.EnqueueRequestsFromMapFunc(r.mapToServicesConfigMap),
+			builder.WithPredicates(nodeconfigChangedPredicate())).
```

Comment:
you can consolidate this into 1 by creating a smarter predicate func

Reply:
please explain why we this watch is needed as well. Do we use node annotations to create (or modify) kubelet args? If so, please point to where we do that because i can't find where in our code